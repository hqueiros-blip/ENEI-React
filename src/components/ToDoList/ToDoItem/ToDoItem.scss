.event {
    &__list {
        display: flex;
        padding: 8px;
        border-bottom: 2px solid #ccc;
        align-items: center;
    }
    &__details {
        width: 60%;
    }
    &__image {
        width: 15%;
        border-radius: 10%;
        object-fit: cover;
        margin-right: 15px;
    }
    &__button {
        margin-top: 10px;
        background-color: #005cad;
        color: #fff;
        border-radius: 20px;
        font-weight: bold;
    }
    &__attended{
        margin-top: 10px;
        font-size: 20px;
        font-weight: bold;
    }
}