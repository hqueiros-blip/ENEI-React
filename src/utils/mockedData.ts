// Mock dos eventos
export const events = [
  {
    id: 1,
    author: "Hugo Queirós",
    description: "React Basics & The Importance of Testing",
    scheduleTime: "9:30 às 11:30",
    image: "src/assets/reactPresentation.jpeg",
    attended: false,
  },
  {
    id: 2,
    author: "Mafalda Landeiro",
    description:
      "Kotlin KickStart: Mastering the Basics for Modern Development",
    scheduleTime: "9:30 às 11:30",
    image: "src/assets/kotlinPresentation.jpeg",
    attended: false,
  },
];
