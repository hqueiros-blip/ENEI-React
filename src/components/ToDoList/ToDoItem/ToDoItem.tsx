import { ToDoItemProps } from "types/toDo";
import "./ToDoItem.scss";

// Recebe o evento e a função de marcar como assistido
export const ToDoItem = ({ eventEnei, onAttend }: ToDoItemProps) => {
  return (
    <li className="event__list">
      {/* Imagem do evento */}
      <img
        src={eventEnei.image}
        alt={eventEnei.author}
        className="event__image"
      />

      {/* Detalhes do evento */}
      <div className="event__details">
        <div>
          <h2>{eventEnei.description}</h2>
        </div>
        <div>
          <h3>{eventEnei.author}</h3>
        </div>
        <div>
          <h4>Horário: {eventEnei.scheduleTime}</h4>
        </div>
        {/* Botão para marcar evento como assistido */}
        {!eventEnei.attended && (
          <button
            onClick={() => onAttend(eventEnei.id)}
            className="event__button"
          >
            Marcar evento como assistido
          </button>
        )}
        {/* Exibe o ícone de check se o evento foi assistido */}
        {eventEnei.attended && (
          <div className="event__attended">
            <span>Evento Assistido ✅</span>
          </div>
        )}
      </div>
    </li>
  );
};
