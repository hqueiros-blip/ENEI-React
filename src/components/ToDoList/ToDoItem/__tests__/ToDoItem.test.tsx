import { render, screen, fireEvent } from "@testing-library/react";
import { ToDoItem } from "../ToDoItem";
import { events } from "utils/mockedData";
import userEvent from "@testing-library/user-event";

const onAttendMock = vi.fn();

describe("ToDoItem component", () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });
  it("Renders the details events", async () => {
    const { asFragment } = render(
      <ToDoItem eventEnei={events[0]} onAttend={onAttendMock} />
    );

    expect(screen.getByText(events[0].description)).toBeInTheDocument();
    expect(screen.getByText(events[0].author)).toBeInTheDocument();
    expect(
      screen.getByText(`Horário: ${events[0].scheduleTime}`)
    ).toBeInTheDocument();
    expect(screen.getByRole("img")).toHaveAttribute("src", events[0].image);

    expect(asFragment()).toMatchSnapshot();
  });

  it("Shows attend button when event is not attended", () => {
    render(<ToDoItem eventEnei={events[0]} onAttend={onAttendMock} />);

    const button = screen.getByRole("button", {
      name: /Marcar evento como assistido/i,
    });
    expect(button).toBeInTheDocument();
  });

  it("Call onAttend function with the correct Id", async () => {
    const user = userEvent.setup();
    render(<ToDoItem eventEnei={events[0]} onAttend={onAttendMock} />);

    const button = screen.getByRole("button");
    await user.click(button);

    expect(onAttendMock).toHaveBeenCalledWith(events[0].id);
  });

  it("Shows attended message when event was attended", () => {
    const attendedEvent = { ...events[0], attended: true };

    render(<ToDoItem eventEnei={attendedEvent} onAttend={onAttendMock} />);

    expect(screen.getByText("Evento Assistido ✅")).toBeInTheDocument();
    expect(screen.queryByRole("button")).not.toBeInTheDocument();
  });
});
