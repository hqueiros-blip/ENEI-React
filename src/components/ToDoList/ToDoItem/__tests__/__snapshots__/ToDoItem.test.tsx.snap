// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ToDoItem component > Renders the details events 1`] = `
<DocumentFragment>
  <li
    class="event__list"
  >
    <img
      alt="Hugo Queirós"
      class="event__image"
      src="/assets/reactPresentation.jpeg"
    />
    <div
      class="event__details"
    >
      <div>
        <h2>
          React Basics & The Importance of Testing
        </h2>
      </div>
      <div>
        <h3>
          Hugo Queirós
        </h3>
      </div>
      <div>
        <h4>
          Horário: 9:30 às 11:30
        </h4>
      </div>
      <button
        class="event__button"
      >
        Marcar evento como assistido
      </button>
    </div>
  </li>
</DocumentFragment>
`;
